test/models/profile_test.dart

```diff
@@ -0,0 +1,185 @@
+        "username": "username",
+        "email": "email",
+        "description": "description",
+        "birth_date": "2021-01-01T00:00:00.000Z",
+        "surname": "surname",
+        "name": "name",
+        "gender": 0,
+        "avatar_url": "avatar_url",
+      };
+      Profile profile = Profile.fromJson(json);
+      expect(profile.id, json["id"]);
+      expect(profile.createdAt, DateTime.parse(json["created_at"]));
+      expect(profile.username, json["username"]);
+      expect(profile.email, json["email"]);
+      expect(profile.description, json["description"]);
+      expect(profile.birthDate, DateTime.parse(json["birth_date"]));
+      expect(profile.surname, json["surname"]);
+      expect(profile.name, json["name"]);
+      expect(profile.gender, Gender.male);
+      expect(profile.avatarUrl, json["avatar_url"]);
+    });
+
+    test('can handle associated models', () async {
+      Map<String, dynamic> json = {
+        "id": 1,
+        "created_at": "2021-01-01T00:00:00.000Z",
+        "username": "username",
+        "email": "email",
+        "reviews_received": ReviewFactory().generateFakeJsonList(length: 3),
+        "profile_features": ProfileFeatureFactory().generateFakeJsonList(length: 3),
+        "reports_received": ReportFactory().generateFakeJsonList(length: 3),
+      };
+      Profile profile = Profile.fromJson(json);
+      expect(profile.reviewsReceived!.length, 3);
+      expect(profile.profileFeatures!.length, 3);
+      expect(profile.reportsReceived!.length, 3);
+    });
+  });
+
+  group('Profile.fromJsonList', () {
+    test('parses a list of profiles from json', () async {
+      Map<String, dynamic> json = {
+        "id": 1,
+        "created_at": "2021-01-01T00:00:00.000Z",
+        "username": "username",
+        "email": "email",
+      };
+
+      List<Map<String, dynamic>> jsonList = [json, json, json];
+      List<Profile> profiles = Profile.fromJsonList(jsonList);
+      expect(profiles.first.id, json["id"]);
+      expect(profiles[1].createdAt, DateTime.parse(json["created_at"]));
+      expect(profiles.last.username, json["username"]);
+    });
+  });
+
+  group('toJson', () {
+    test('returns a json representation of the profile', () async {
+      Profile profile = ProfileFactory().generateFake();
+      Map<String, dynamic> json = profile.toJson();
+      expect(json["username"], profile.username);
+      expect(json["email"], profile.email);
+      expect(json["description"], profile.description);
+      expect(json["birth_date"], profile.birthDate!.toString());
+      expect(json["surname"], profile.surname);
+      expect(json["name"], profile.name);
+      expect(json["gender"], profile.gender?.index);
+      expect(json["avatar_url"], profile.avatarUrl);
+    });
+  });
+
+  group('toString', () {
+    test('returns a string representation of the profile', () async {
+      Profile profile = ProfileFactory().generateFake(
+        id: 1,
```

Comment:
what happens if id or created_at is null?

Reply:
I also think this is an edgecase that should not really happen in normal usage. Also, the ModelFactory pattern currently does not support nullable id and created_at yet (we could add that but probably not worth it).

Reply:
i guess same reasoning as above that it's not a relevant case for us.